filebeat/module/apache2/error/ingest/pipeline.json

```diff
@@ -5,27 +5,15 @@
       "grok": {
         "field": "message",
         "patterns": [
-          "\\[%{APACHE_TIME:apache2.error.timestamp}\\] \\[%{LOGLEVEL:apache2.error.level}\\]( \\[client %{IPORHOST:apache2.error.client}\\])? %{GREEDYDATA:apache2.error.message}",
-          "\\[%{APACHE_TIME:apache2.error.timestamp}\\] \\[%{DATA:apache2.error.module}:%{LOGLEVEL:apache2.error.level}\\] \\[pid %{NUMBER:apache2.error.pid}(:tid %{NUMBER:apache2.error.tid})?\\]( \\[client %{IPORHOST:apache2.error.client}\\])? %{GREEDYDATA:apache2.error.message1}"
+          "\\[%{APACHE_TIME:apache2.error.timestamp}\\] \\[%{LOGLEVEL:log.level}\\]( \\[client %{IPORHOST:source.address}\\])? %{GREEDYDATA:message}",
+          "\\[%{APACHE_TIME:apache2.error.timestamp}\\] \\[%{DATA:apache2.error.module}:%{LOGLEVEL:log.level}\\] \\[pid %{NUMBER:process.pid:int}(:tid %{NUMBER:process.thread.id:long})?\\]( \\[client %{IPORHOST:source.address}\\])? %{GREEDYDATA:message}"
```

Comment:
@webmat To my surprise I had to use `:long` here as it seems for `:int` the value was too big. Probably good to know if you hit issue with other values too and we should be careful as we now know there is a limit.

@jakelandis Is there any disadvantage here to always use `:long` instead of `:int`?

Reply:
Wow I'm glad we caught the `:int` case with our test data! I've been using only "int" for all of the coercions I've fixed so far. Although all of those I did fit in the bounds, AFAICT (ports, PIDs, mostly).

I think always using `long` would actually be fine. It'll instantiate a long, but then shove that into a JSON object, where the distinction disappears anyway.

I'm now curious to know if the `int` used by grok is signed or not. Both possibilities can actually cause problems. If it's signed, it's currently not enough to properly capture the high ports or PIDs, over 32k. If it's unsigned, it wouldn't work to capture negative integers...

I created #9598 to tackle this.

Reply:
@ruflin - no disadvantage other then you may also need to update the mapping.

@webmat - It is signed. Specifically we use https://docs.oracle.com/javase/8/docs/api/java/lang/Long.html#parseLong-java.lang.String- or https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html#parseInt-java.lang.String- to cast the value found from the grok expression.

Reply:
Great, thanks for confirming, @jakelandis.

Most of our fields actually are `long` in ElasticSearch, so this won't be a problem. I'll fix our coercions soon.